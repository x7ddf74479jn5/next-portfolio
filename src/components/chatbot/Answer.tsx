import Button from "@material-ui/core/Button";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import React from "react";
import type { SelectAnswer } from "src/types/chatbot";
import type { Answer as AnswerType } from "src/types/chatbot";

const useStyles = makeStyles(() => {
  return createStyles({
    button: {
      borderColor: "#ffb549",
      color: "#ffb549",
      fontWeight: 600,
      marginBottom: "8px",
      "&:hover": {
        backgroundColor: "#ffb549",
        color: "#fff",
      },
    },
  });
});

type Props = AnswerType & {
  select: SelectAnswer;
  isSelectReady: () => boolean;
};

const Answer: React.VFC<Props> = (props) => {
  const classes = useStyles();
  const handleClick = () => {
    if (!props.isSelectReady()) {
      return;
    }

    return props.select(props.content, props.nextId);
  };

  return (
    <Button className={classes.button} variant="outlined" onClick={handleClick}>
      {props.content}
    </Button>
  );
};
export default Answer;
