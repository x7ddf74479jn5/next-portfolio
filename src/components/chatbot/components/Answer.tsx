import Button from "@material-ui/core/Button";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import React from "react";
import type { SelectAnswer } from "src/types/chatbot";
import { Answer } from "src/types/chatbot";

const useStyles = makeStyles(() => {
  return createStyles({
    button: {
      borderColor: "#ffb549",
      color: "#ffb549",
      fontWeight: 600,
      marginBottom: "8px",
      "&:hover": {
        backgroundColor: "#ffb549",
        color: "#fff",
      },
    },
  });
});

type Props = Answer & {
  select: SelectAnswer;
};

const Answer: React.VFC<Props> = (props) => {
  const classes = useStyles();

  return (
    <Button
      className={classes.button}
      variant="outlined"
      onClick={() => {
        return props.select(props.content, props.nextId);
      }}
    >
      {props.content}
    </Button>
  );
};
export default Answer;
