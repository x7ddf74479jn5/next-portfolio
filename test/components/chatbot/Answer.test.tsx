import userEvent from "@testing-library/user-event";
import Answer from "src/components/chatbot/Answer";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Answer", () => {
  let renderResult: RenderResult;
  let buttonElement: HTMLElement;
  const mockHandleSelect = jest.fn();
  const mockHandleIsSelectReady = jest.fn().mockReturnValue(true);

  beforeEach(() => {
    renderResult = render(
      <Answer content="content" nextId="nextId" select={mockHandleSelect} isSelectReady={mockHandleIsSelectReady} />
    );
    buttonElement = renderResult.getByRole("button");
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(buttonElement).toHaveTextContent("content");
    expect(buttonElement).toBeEnabled();
  });

  it("user interaction", () => {
    userEvent.click(buttonElement);
    expect(mockHandleIsSelectReady).toHaveBeenCalled();
    expect(mockHandleSelect).toHaveBeenCalled();
  });
});
