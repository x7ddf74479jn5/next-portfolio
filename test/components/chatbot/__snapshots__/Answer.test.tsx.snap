// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      content
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
